CONTRIBUTING.md

```diff
@@ -1,3 +1,119 @@
 # License
 
 By contributing your code, you agree to license your contribution under the MIT License.
+
+You also need to sign the [CLA](https://cla-assistant.io/bountysource/core) before your Pull Request is accepted.
+When you send your PR check the Pull Request page, it wil have a Link to signing the CLA.
+
+# How to Contribute
+Keep in mind this standardized setup is a work in progress, if you have any trouble please open an issue on the issue tracker and give as much info as possible.
+
+First of all you need [Docker](https://www.docker.com/what-docker) and [Docker Compose](https://docs.docker.com/compose/install/) installed.
+
+## Environment
+Bountysource uses Environment Variables for a lot of things. There's a suggested `.env.dev` file at the root directory that is loaded by the `docker-compose.yml`.
+
+It's recommended that you take a good look at the entire `.env.dev` file and, if necessary, change it to suit your needs, but at the very least you should pay attention to the following variables:
+
+0. URL for Buntysource API: `BOUNTYSOURCE_API_URL`
+0. URL for Bountysource Website (Front-End): `BOUNTYSOURCE_WWW_URL`
+0. URL for the Database: `DATABASE_URL`
+0. Environment for Rails and Rake: `RAILS_ENV` and `RAKE_ENV`, respectively.
+
+### About the `DATABASE_URL`
+This is a url, in the usual form `protocol://user:password@host/path`, just without the `password` since the postgres image doesn't use a password.
+The `path` part is the database name and you can change that without any issues, but before changing anything else keep in mind that the `host` matches the service in the `docker-compose.yml` and `docker-compose-test.yml` files, and the `user` matches the default on the `postgres` docker image.
+``` bash
+DATABASE_URL=postgres://postgres@pgsql/bountysource_dev
+```
+
+### The URLs
+Remember to keep the `API` and `WWW` URLs accessible from Outside Docker, either by changing your `hosts` file or leaving them as `0.0.0.0`:
+``` bash
+BOUNTYSOURCE_API_URL=http://0.0.0.0:3000/
+BOUNTYSOURCE_WWW_URL=http://0.0.0.0:3000/
+```
+
+## Standardized Setup for First-time use
+
+There is some setup and configuration you need to do before the project will run, this section covers those.
+
+Mainly, you need to create the Sphinx configuration file as well as Symlink it to its expected location before running the Search Daemon (Default command Sphinx image runs). You also have to generate the database and search indexes.
+
+0. Build container and start database.  
```

Comment:
I'm taking a look at `foreman` and updated the PR with the improved workflow for Docker so we have a working alternative.

Foreman would also require manually managing versions and dependencies, which, after a few years of managing servers and using a Rolling Release Linux distro, I try to avoid at all costs. It may also lead to bugs due to differences in developer environments.

We could also provide both options, so each developer can decide. Do you think it's a good idea, or would you rather completely remove Docker?